### PR TITLE
bpo-30459: Cast the result of PyCell_SET to void

### PR DESCRIPTION
**Pull request by vstinner**
https://bugs.python.org/issue30459

### CODE REVIEW

File: Doc/whatsnew/3.10.rst [hunk 1]
```diff
@@ -611,6 +611,10 @@ Porting to Python 3.10
   :ref:`Python Path Configuration.  <init-path-config>`.
   (Contributed by Victor Stinner in :issue:`42260`.)
 
+* Cast the result of :c:func:`PyList_SET_ITEM`, :c:func:`PyTuple_SET_ITEM` and
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.10.rst]: I would say that PyList_SET_ITEM etc are no longer rvalues and have type void. Or something like.

Casting to void is an implementation detail. What is important to the user that they can no longer write `x = PyList_SET_ITEM(a, b, c)` or `PyList_SET_ITEM(a, b, c) = x`.

**Comment by vstinner** [comment]
@serhiy-storchaka: I rewrote the NEWS entry/What's New entry. Is it better?

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by vstinner** [comment]
Thanks for the review @serhiy-storchaka!

**Comment by serhiy-storchaka** [comment]
Thank you for catching the case of PyCell_SET. I missed it.

And this change was worth to be documented.

**Comment by vstinner** [comment]
> And this change was worth to be documented.

My experience with rebuilding Fedora with a newer Python version is that every single minor change breaks at least one Python package, and it's a pain to find which change broke the package, and then find information about the change. IMO it's a good practice to document changes which are backward incompatible on purpose, even if the new behavior was the documented behavior.

**Pull request merged**